src/gui/guiChatConsole.cpp

```diff
@@ -328,6 +353,16 @@ void GUIChatConsole::drawText()
 			core::rect<s32> destrect(
 				x, y, x + m_fontsize.X * fragment.text.size(), y + m_fontsize.Y);
 
+
+			core::recti rect;
+			if (m_scrollbar_enabled && m_scrollbar->isVisible()) {
+				// leave 4 pixels of space between scrollbar and text
+				rect = core::rect<s32> (0, 0, m_screensize.X - 32 - 4, m_height);
+				m_scrollbar->setPos(buf.getScrollPosition());
```

Comment:
I am aware this is in a loop even though it really shouldn't, I'll change that before I remove the WIP status